src/com/hms_networks/americas/sc/realtimedata/RealTimeDataQueueManager.java

```diff
@@ -126,9 +130,12 @@ public static DataPoint getGroupNextData(int tagGroup) {
    */
   public static int getNumGroupDataPoints(int tagGroup) {
     int numDataPoints = 0;
-    for (int i = 0; i < ((ArrayList) tagManagers.get(tagGroup)).size(); i++) {
-      numDataPoints +=
-          ((RealTimeTagDataPointManager) ((ArrayList) tagManagers.get(tagGroup)).get(i)).getSize();
+    if (((ArrayList) tagManagers.get(tagGroup)) != null) {
+      for (int i = 0; i < ((ArrayList) tagManagers.get(tagGroup)).size(); i++) {
+        numDataPoints +=
+            ((RealTimeTagDataPointManager) ((ArrayList) tagManagers.get(tagGroup)).get(i))
+                .getSize();
+      }

```

Comment:
There are a lot of repeat calls to `tagManagers.get(tagGroup)`. They should be consolidated where possible.

Reply:
Since ArrayLists are backed by an array, get() is a constant O(1) operation, so the repeat calls don't add much time complexity, but it looks neater to have a variable for storing `tagManagers.get(tagGroup)`.

Reply:
yeah agreed. I'll add a commit for this since it's a bit outside the scope of the current commits